File: curriculum/algos/bubble_sort.py
def bubble_sort(values):
	boundary = len(values)
	while boundary > 1:
		changed = False
		for index in range(1, boundary):
			if values[index] < values[index - 1]:
				values[index - 1], values[index] = values[index], values[index - 1]
				changed = True
		if not changed:
			break
		boundary -= 1

File: curriculum/algos/bubble_sort.rs
pub fn bubble_sort(values: &mut [i32]) {
let mut boundary = values.len();
while boundary > 1 {
let mut changed = false;
for index in 1..boundary {
if values[index] < values[index - 1] {
values.swap(index - 1, index);
changed = true;
}
}
if !changed { break; }
boundary -= 1;
}
}
